### PR TITLE
Add some unit tests on number formatting (#528)

### PR DESCRIPTION
**Pull request by galak75**
As discussed in #528

### CODE REVIEW

File: tests/test_collection.py [hunk 1]
```diff
@@ -349,6 +349,76 @@ def test_no_filter(self):
...
+        with fiona.open(self.filename, driver=driver, encoding='utf-8') as c:
+            self.assertEqual(len(c), 2)
+
+            rf1, rf2 = list(c)
+
+            # integers are truncated
+            self.assertEqual(rf1['properties']['integer'], 12)
+            self.assertEqual(rf2['properties']['integer'], 12)
+
+            # floats are rounded
```
Comment (sgillies) [inline tests/test_collection.py]: @galak75 would you be willing to make the float assertions into a separate test function? The truncation and rounding effects have different causes.

File: tests/test_collection.py [hunk 2]
```diff
@@ -349,6 +349,76 @@ def test_no_filter(self):
...
+            self.assertEqual(len(c), 2)
+
+            rf1, rf2 = list(c)
+
+            # integers are truncated
+            self.assertEqual(rf1['properties']['integer'], 12)
+            self.assertEqual(rf2['properties']['integer'], 12)
+
+            # ****************************************
+            # FLOAT FORMATTING IS NOT RESPECTED...
```
Comment (sgillies) [inline tests/test_collection.py]: Here we should also have a different test since there is a split between expected and unexpected results.

**Comment by galak75** [comment]
@sgillies thank you for your comments. I split assertions in separate tests. Feel free to suggest different test function names (English is not my mother tongue), or additional changes.

Also, I added some new tests, based on [your comment](https://github.com/Toblerity/Fiona/issues/528#issuecomment-355327527) : 

> The integer conversion is done by Fiona and only exists to trigger exceptions when we mistakenly try to copy non-numeric values to numeric fields

- `"12.22"` is converted to `12.22` float value before being written by the target driver
- `"invalid number"` is converted to `0`

But there is actually no exception (which might be the correct behavior)

These new tests might be not necessary if already tested else where. I can remove them if you prefer. Just let me know.

**Comment by sgillies** [comment]
Thanks @galak75 . I'm fixing up the Windows builds and will see about merging this after AppVeyor gives us the green light.

File: tests/test_collection.py [hunk 3]
```diff
@@ -392,15 +392,22 @@ def test_shape_file_properties_number_formatting(self):
...
-            # floats are rounded
+    def test_shape_driver_rounds_float_property_to_requested_digits_number(self):
+        driver = "ESRI Shapefile"
+        self._write_collection(driver)
+
+        with fiona.open(self.filename, driver=driver, encoding='utf-8') as c:
+            self.assertEqual(len(c), 2)
+
+            rf1, rf2 = list(c)
+
```
Comment (sgillies) [inline tests/test_collection.py]: Excellent !

**Comment by galak75** [comment]
Glad to see that ! these are only tests, but it is my first contribution to an open source project. :smiley:

**Pull request merged**